File: ParkPal.Application/src/models/api/AttractionTimer.ts
export default class AttractionTimer {
    attractionTimerId?: number;
    minuteInterval?: number;
    parkId?: string;
    attractionId?: string;
    criteriaType?: number;
    waitTime = 0;
    enabled?: boolean;

    constructor(data: Pick<AttractionTimer, "attractionTimerId" | "minuteInterval" | "parkId" | "attractionId" | "criteriaType" | "waitTime" | "enabled"> | null = null) {
        if(data != null) {
            this.attractionTimerId = data.attractionTimerId;
            this.minuteInterval = data.minuteInterval;
            this.parkId = data.parkId;
            this.attractionId = data.attractionId;
            this.criteriaType = data.criteriaType;
            this.waitTime = data.waitTime;
            this.enabled = data.enabled;
        }
    }
}

File: ParkPal.Common/Models/Database/Entities/Notification/AttractionTimer.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Collections.Generic;
using ParkPal.Common.Models.Database.Entities.Device;

namespace ParkPal.Common.Models.Database.Entities.Notification
{
    [Obsolete("AttractionTimer is due to be removed and is therefore obsolete, please use Notification.Item going forwards.")]
    [Table("AttractionTimer", Schema = "Notification")]
    public class AttractionTimer
    {
        [Key]
        public int AttractionTimerId { get; set; }
        
        public int MinuteInterval { get; set; }
        
        [MaxLength(1000)]
        public string ParkId { get; set; }
        
        [MaxLength(1000)]
        public string AttractionId { get; set; }
        
        public int CriteriaType { get; set; }
        
        public int WaitTime { get; set; }
        
        public bool Enabled { get; set; }
        
        [ForeignKey("Subscription")]
        public int SubscriptionId { get; set; }
        public Subscription Subscription { get; set; }
    }
}